src/functions-and-arrays.js

```diff
@@ -1,73 +1,488 @@
+    sum += array[0];
+    return sum;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      if (array[i] === 0) {
+        sum = 0;
+      } else {
+        sum += array[i];
+      }
+    }
+    return sum;
+  }
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  let average = 0;
+  let sum = 0;
+  if (array.length == 0) {
+    return null;
+  } else if (array.length === 1) {
+    average += array[0];
+    return average;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      sum += array[i];
+    }
+    average = sum / array.length;
+    return average;
+  }
+}
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
+
+function averageWordLength(array) {
+  let average = 0;
+  let sum = 0;
+  if (array.length === 0) {
+    return null;
+  } else if (array.length === 1) {
+    average = array[0].length;
+    return average;
+  } else {
+    for (let i = 0; i < array.length; i++) {
+      sum += array[i].length;
+    }
+    average = sum / array.length;
+    return average;
+  }
+}
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(array) {
+  if (array.length === 0) {
+    return null;
+  } else {
+    let newArray = [];
+    for (let i = 0; array.indexOf(array[i]) != -1; i++) {
+      newArray.push(array.indexOf(array[i]));
+      idx = array.indexOf(array[i], array.indexOf(array[i]) + 1);
+    }
+    return newArray;
+  }
+}
+
+let test = uniquifyArray(wordsUnique);
+
+console.log(test);
+
 // Iteration #6: Find elements
-const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
+const wordsFind = [
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
+];
+
+function doesWordExist(array, word) {
+  if (array.length === 0) {
+    return null;
+  } else if (array.includes(word)) {
+    return true;
+  } else {
+    return false;
+  }
+}
 
 // Iteration #7: Count repetition
 const wordsCount = [
-  'machine',
-  'matter',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'matter',
-  'truth',
-  'disobedience',
-  'matter'
+  "machine",
+  "matter",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "matter",
+  "truth",
+  "disobedience",
+  "matter"
 ];
 
+function howManyTimes(array, word) {
```

Comment:
Good progress on this iteration! 
- You could use the || operator if the first condition and do ( array.length === 0 || !array.includes(word) ) since they both return 0.
- for the other two checks, you could create a counter variable and do a for loop or forEach that adds 1 everytime the word being checked inside the loop is the same as the word passed as an argument in the function.